utils/replicaset/canary.go

```diff
@@ -279,6 +287,25 @@ func GetCurrentSetWeight(rollout *v1alpha1.Rollout) int32 {
 	return 0
 }
 
+func UseSetCanaryScale(rollout *v1alpha1.Rollout) (v1alpha1.SetCanaryScale, bool) {
```

Comment:
1. Style: instead of return value of `(v1alpha1.SetCanaryScale, bool)`, which means we would sometimes return an empty struct, we could instead return a pointer (*v1alpha1.SetCanaryScale), where nil obviously representing that it should be used.

2. This method should return false (or nil per my previous suggestion) if the rollout spec.strategy.canary.trafficRouting is nil. This is because the `setCanaryScale` should not be used with the regular replica-weighted canary, since that strategy can only use matchWeight.